### ISSUE TITLE
Can't input text in TextField.

### ISSUE THREAD

**Issue by monxarat** [flutter/flutter#27331]
When I touch in the TextField and type character -> suggestion show immediately dismiss and cursor moving to first text. I can't input any text.

If set `_textEditingController` to `TextField` above problem has occurred, if not set `_textEditingController`, the input text is ok(But can't get text has input)

```dart
TextEditingController _textEditingController = TextEditingController();

  @override
  void initState() {
    _textEditingController.addListener(() {
      print("CONTROLLER: $_textEditingController");
    });
    super.initState();
  }

Widget _buildTextFieldMemo() {
    return TextField(
      controller: _textEditingController,
      textInputAction: TextInputAction.done,
      onChanged: (String val) {
        setState(() {
          _isValid = _validate();
        });
      },
      keyboardType: TextInputType.multiline,
      maxLines: 4,
    );
  }

```

![device-2019-01-31-122845](https://user-images.githubusercontent.com/4075787/52028889-30ef9900-2554-11e9-9f4c-6e547bab4f08.png)

Logs.
```
I/flutter (30849): CONTROLLER: TextEditingController#39911(TextEditingValue(text: ┤đồ ăn sáng.├, selection: TextSelection(baseOffset: 11, extentOffset: 11, affinity: TextAffinity.upstream, isDirectional: false), composing: TextRange(start: -1, end: -1)))
I/HwSecImmHelper(30849): mSecurityInputMethodService is null
I/HwSecImmHelper(30849): mSecurityInputMethodService is null
I/flutter (30849): CONTROLLER: TextEditingController#39911(TextEditingValue(text: ┤đồ ăn sáng.├, selection: TextSelection(baseOffset: 11, extentOffset: 11, affinity: TextAffinity.downstream, isDirectional: false), composing: TextRange(start: -1, end: -1)))
```

```
W/IInputConnectionWrapper(30849): beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): setComposingText on inactive InputConnection
W/IInputConnectionWrapper(30849): endBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): finishComposingText on inactive InputConnection
W/IInputConnectionWrapper(30849): beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): endBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): endBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): endBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper(30849): setComposingText on inactive InputConnection
W/IInputConnectionWrapper(30849): endBatchEdit on inactive InputConnection
```

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

To investigate we would need a minimal runnable reproduction
as a single file so that we can just copy your code into `lib/main.dart` of a new project and run to reproduce.

**Comment by monxarat** [comment]
/home/chinhnq/Downloads/flutter/flutter_linux_v1.0.0-stable/flutter/bin/flutter doctor --verbose
[✓] Flutter (Channel stable, v1.0.0, on Linux, locale en_US.UTF-8)
    • Flutter version 1.0.0 at /home/chinhnq/Downloads/flutter/flutter_linux_v1.0.0-stable/flutter
    • Framework revision 5391447fae (9 weeks ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[!] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /home/chinhnq/Android/Sdk
    • Android NDK at /home/chinhnq/Android/Sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /home/chinhnq/Android/Sdk
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Android Studio (version 3.2)
    • Android Studio at /opt/android-studio-3.2
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b04)

[✓] Android Studio (version 3.3)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 31.3.3
    • Dart plugin version 182.5124
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] VS Code (version 1.30.2)
    • VS Code at /usr/share/code
    • Flutter extension version 2.22.2

[✓] Connected device (1 available)
    • WAS LX2J • P3PDU18329002411 • android-arm64 • Android 7.0 (API 24)

! Doctor found issues in 1 category.
Process finished with exit code 0

**Comment by zoechi** [comment]
> To investigate we would need a minimal runnable reproduction
as a single file so that we can just copy your code into lib/main.dart of a new project and run to reproduce.

Can you please make a runnable `lib/main.dart` from your code snippets?

**Comment by monxarat** [comment]
I have written some codes, on lib/main.dart. Code run is ok. I have realized my problem in my code when I set the text into `_textEditingController` in the `FutureBuilder`. each time input the character, `FutureBuilder` will be reloaded with `connectionState done`. How to avoid it?

My code
```
...
FutureBuilder(
        future: DBProvider.db.getProduct(productID),
        builder: (BuildContext context, AsyncSnapshot<List<Product>> products) {
          if (products.hasData) {
            product = products.data[0];
            _textEditingController.text = product.name;

            return _createBodyProduct();
          }
        })
...
```

**Comment by jonahwilliams** [comment]
@monxarat See the [FutureBuilder](https://docs.flutter.io/flutter/widgets/FutureBuilder-class.html) documentation:

> The future must have been obtained earlier, e.g. during State.initState, State.didUpdateConfig, or State.didChangeDependencies. It must not be created during the State.build or StatelessWidget.build method call when constructing the FutureBuilder. If the future is created at the same time as the FutureBuilder, then every time the FutureBuilder's parent is rebuilt, the asynchronous task will be restarted.

TLDR: move `DBProvider.db.getProduct(productID)` into a State lifecycle.
